src/main/java/org/fisco/bcos/web3j/crypto/ECDSASigner.java

```diff
@@ -0,0 +1,266 @@
+     *
+     * @param kCalculator a K value calculator.
+     */
+    public ECDSASigner(DSAKCalculator kCalculator) {
+        this.kCalculator = kCalculator;
+    }
+
+    @Override
+    public void init(boolean forSigning, CipherParameters param) {
+        SecureRandom providedRandom = null;
+
+        if (forSigning) {
+            if (param instanceof ParametersWithRandom) {
+                ParametersWithRandom rParam = (ParametersWithRandom) param;
+
+                this.key = (ECPrivateKeyParameters) rParam.getParameters();
+                providedRandom = rParam.getRandom();
+            } else {
+                this.key = (ECPrivateKeyParameters) param;
+            }
+        } else {
+            this.key = (ECPublicKeyParameters) param;
+        }
+
+        this.random =
+                initSecureRandom(forSigning && !kCalculator.isDeterministic(), providedRandom);
+    }
+
+    // 5.3 pg 28
+    /**
+     * generate a signature for the given message using the key we were initialised with. For
+     * conventional DSA the message should be a SHA-1 hash of the message of interest.
+     *
+     * @param message the message that will be verified later.
+     */
+    @Override
+    public BigInteger[] generateSignature(byte[] message) {
+        ECDomainParameters ec = key.getParameters();
+        BigInteger n = ec.getN();
+        BigInteger e = calculateE(n, message);
+        BigInteger d = ((ECPrivateKeyParameters) key).getD();
+
+        if (kCalculator.isDeterministic()) {
+            kCalculator.init(n, d, message);
+        } else {
+            kCalculator.init(n, random);
+        }
+
+        BigInteger r, s;
+
+        ECMultiplier basePointMultiplier = createBasePointMultiplier();
+
+        // 5.3.2
+        do // generate s
+        {
+            BigInteger k;
+            do // generate r
+            {
+                k = kCalculator.nextK();
+
+                ECPoint p = basePointMultiplier.multiply(ec.getG(), k).normalize();
+
+                // 5.3.3
+                r = p.getAffineXCoord().toBigInteger().mod(n);
+            } while (r.equals(ZERO));
+
+            s = k.modInverse(n).multiply(e.add(d.multiply(r))).mod(n);
+        } while (s.equals(ZERO));
+
+        return new BigInteger[] {r, s};
+    }
+
+    /**
+     * The same generateSignature with the temporary variable ECPoint P generated by the signature
+     * process is also returned together
+     *
+     * @param message the message that will be verified later.
+     */
+    public Object[] generateSignature2(byte[] message) {
+        ECDomainParameters ec = key.getParameters();
+        BigInteger n = ec.getN();
+        BigInteger e = calculateE(n, message);
+        BigInteger d = ((ECPrivateKeyParameters) key).getD();
+
+        if (kCalculator.isDeterministic()) {
+            kCalculator.init(n, d, message);
+        } else {
+            kCalculator.init(n, random);
+        }
+
+        BigInteger r, s;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/fisco/web3sdk/pullRequest?prid=4982666)